ios/Mappers.swift

```diff
@@ -459,15 +461,18 @@ class Mappers {
         guard let shippingDetails = shippingDetails else {
             return nil
         }
-        let shippingAddress = STPPaymentIntentShippingDetailsAddressParams(line1: shippingDetails["addressLine1"] as? String ?? "")
-
-        shippingAddress.city = shippingDetails["addressCity"] as? String
-        shippingAddress.postalCode = shippingDetails["addressPostalCode"] as? String
-        shippingAddress.country = shippingDetails["addressCountry"] as? String
-        shippingAddress.line1 = shippingDetails["addressLine1"] as? String ?? ""
-        shippingAddress.line2 = shippingDetails["addressLine2"] as? String
-        shippingAddress.state = shippingDetails["addressState"] as? String
+        
+        let shippingAddress = STPPaymentIntentShippingDetailsAddressParams(line1: "")
 
+        if let addressMap = shippingDetails["address"] as? NSDictionary {
+            shippingAddress.city = addressMap["city"] as? String
+            shippingAddress.postalCode = addressMap["postalCode"] as? String
```

Comment:
If people are sloppy on the RN side, they might provide postalCode as a number (Particularly American-centric UIs). Do you need to support NSNumber as an option here?

Reply:
You'll get a redscreen error with a clear error message if you try to do that (rather than just have a missing postal code) so we should be fine.